src/index.ts

```diff
@@ -545,6 +545,11 @@ prog
   .example('lint src test --fix')
   .option('--ignore-pattern', 'Ignore a pattern')
   .example('lint src test --ignore-pattern test/foobar.ts')
+  .option(
+    '--max-warnings',
+    'Exits with non-zero error code if warnings exceed this number'
```

Comment:
```suggestion
    'Exits with non-zero error code if number of warnings exceed this number',
    Infinity
```

sets the default to go along with the interface below